httpdindex patch generated by KarLS Builder Infrastructure
make httpd use interpreted cgis as index files
diff -ru --no-dereference src/busybox-1.34.1.orig/busybox-1.34.1/networking/httpd.c src/busybox-1.34.1/networking/httpd.c
--- src/busybox-1.34.1.orig/busybox-1.34.1/networking/httpd.c	2021-06-16 12:02:16.000000000 +0200
+++ src/busybox-1.34.1/networking/httpd.c	2022-10-01 23:31:30.812620570 +0200
@@ -1583,8 +1583,15 @@
 
 	/* SCRIPT_FILENAME is required by PHP in CGI mode */
 	if (home_httpd[0] == '/') {
-		char *fullpath = concat_path_file(home_httpd, url);
-		setenv1("SCRIPT_FILENAME", fullpath);
+		if (url[strlen(url)-1]=='/') {
+			char *fullpath, *realurl = malloc(sizeof(url)+sizeof(index_page)+1);
+			sprintf(realurl, "%s%s", url, index_page);
+			fullpath = concat_path_file(home_httpd, realurl);
+			setenv1("SCRIPT_FILENAME", fullpath);
+		} else {
+			char *fullpath = concat_path_file(home_httpd, url);
+			setenv1("SCRIPT_FILENAME", fullpath);
+		}
 	}
 	/* set SCRIPT_NAME as full path: /cgi-bin/dirs/script.cgi */
 	setenv1("SCRIPT_NAME", url);
@@ -1665,6 +1672,10 @@
 		}
 		script++;
 
+		if (strcmp(script, "") == 0) {
+			script = index_page;
+		}
+
 		/* set argv[0] to name without path */
 		argv[0] = script;
 		argv[1] = NULL;
